timeseries/src/autogluon/timeseries/models/statsmodels/abstract_statsmodels.py

```diff
@@ -0,0 +1,175 @@
+        raise NotImplementedError
+
+    def _fit(self, train_data: TimeSeriesDataFrame, time_limit: int = None, **kwargs):
+        # Select timeseries that don't have a fitted local model
+        train_hash = hash_ts_dataframe_items(train_data)
+        items_to_fit = [item_id for item_id, ts_hash in train_hash.iteritems() if ts_hash not in self._fitted_models]
+        timeseries_to_fit = (train_data.loc[item_id][self.target] for item_id in items_to_fit)
+        default_model_init_args, default_model_fit_args = self._get_default_model_init_and_fit_args()
+        # TODO: Should we rather set n_jobs in __init__?
+        self.n_jobs = default_model_fit_args.pop("n_jobs", -1)
+
+        # Fit models in parallel
+        fit_fn = partial(
+            self._fit_local_model,
+            default_model_fit_args=default_model_fit_args,
+            default_model_init_args=default_model_init_args,
+        )
+        fitted_models = Parallel(n_jobs=self.n_jobs)(delayed(fit_fn)(timeseries=ts) for ts in timeseries_to_fit)
+        for item_id, model in zip(items_to_fit, fitted_models):
+            self._fitted_models[train_hash.loc[item_id]] = model
+
+    def predict(self, data: TimeSeriesDataFrame, quantile_levels: List[float] = None, **kwargs) -> TimeSeriesDataFrame:
+        self._check_predict_inputs(data, quantile_levels=quantile_levels, **kwargs)
+        if quantile_levels is None:
+            quantile_levels = self.quantile_levels
+
+        # Make sure that we fitted a local model to each time series in data
+        data_hash = hash_ts_dataframe_items(data)
+        items_to_fit = [item_id for item_id, ts_hash in data_hash.iteritems() if ts_hash not in self._fitted_models]
+        if len(items_to_fit) > 0:
+            logger.info(f"{self.name} received {len(items_to_fit)} items not seen during training, re-running fit")
+            self._fit(train_data=data)
+
+        # Make predictions in parallel
+        predict_fn = partial(self._predict_with_local_model, quantile_levels=quantile_levels)
+        timeseries_with_models = (
+            (data.loc[item_id][self.target], self._fitted_models[ts_hash])
+            for item_id, ts_hash in data_hash.iteritems()
+        )
+        predictions = Parallel(n_jobs=self.n_jobs)(
+            delayed(predict_fn)(timeseries=ts, fitted_model=model) for ts, model in timeseries_with_models
+        )
+
+        # Combine all predictions into a single DataFrame
+        result_df = pd.concat({item_id: pred for item_id, pred in zip(data.iter_items(), predictions)})
+        result_df.index.rename([ITEMID, TIMESTAMP], inplace=True)
+        return TimeSeriesDataFrame(result_df)
+
+    def _predict_with_local_model(
+        self, timeseries: pd.Series, fitted_model: FittedLocalModel, quantile_levels: List[float]
+    ) -> pd.DataFrame:
+        raise NotImplementedError
+
+    def _get_predictions_from_initialized_model(self, initialized_model, cutoff, quantile_levels):
```

Comment:
nit: is the word `initialized` redundant here? should it be something like `sm_model` or `internal_model` instead?

Reply:
also, could we add type hints perhaps?

Reply:
I think `initialized` is slightly more appropriate here since we initialize the internal intermediate results inside the model that are necessary for prediction (by calling `fit` with `maxiter=0`), but this is not a strong opinion, we can change the name if you think it improves the readability.

Will add the type hints.

Reply:
Up to you. I think one of the main difficulties we have in naming is that the word `model` is overloaded. It refers to both our model, and the SM model. I thought `initialized` was redundant as you can't really get predictions from a model that's not initialized. Up to you, no strong opinions as well.

Reply:
I agree, changes this to `sm_model` and the method name to `_get_predictions_from_statsmodels_model`